### PR TITLE
Update TERMS.md with definition for Setting

### PR DESCRIPTION
**Pull request by cwillum**
Signed-off-by: cwillum <cwmmoore@amazon.com>

### Description
Added a definition for Setting.

### Issues Resolved
Fix for #1631 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TERMS.md [hunk 1]
```diff
@@ -544,6 +544,12 @@ A set of conditions, internals, and actions that create notifications.
 
 Avoid using. Use *separate* or *isolate* instead.
 
+**setting**
+
+Refers to the key-value pairs that create mappings in the various YAML configuration files used throughout OpenSearch. Sometimes alternatively called parameters, the programming language manipulating the key-value pair usually dictates the name of this mapping in a YAML file. For OpenSearch documentation (java) they are properly a `Setting` object.
```
Comment (alicejw-aws) [inline TERMS.md:549]: a couple of optional suggestions...

"A key-value pair that creates a mapping in the . . . "
"For example, with Java, you can create a`Settings` object that takes a key-value pair, such as `Settings.index.number_of_shards` set to a value of 4."

**Review by alicejw-aws — APPROVED** [review]
Looks great!

**Review by vagimeli — APPROVED** [review]

**Comment by vagimeli** [comment]
LGTM

File: TERMS.md (hunk 1 above)
Comment (vagimeli) [reply TERMS.md:549]: +1

**Review by natebower — COMMENTED** [review]
@cwillum Just a couple of minor changes. Thanks!

File: TERMS.md [hunk 2]
```diff
@@ -544,6 +544,16 @@ A set of conditions, internals, and actions that create notifications.
 
 Avoid using. Use *separate* or *isolate* instead.
 
+**setting**
+
+A key-value pair that creates a mapping in one of the many YAML configuration files used throughout OpenSearch. Sometimes alternatively called parameters, the programming language manipulating the key-value pair usually dictates the name of this mapping in a YAML file. For OpenSearch documentation (Java) they are properly a `Setting` object.
```
Comment (natebower) [inline TERMS.md:549]:
```suggestion
A key-value pair that creates a mapping in one of the many YAML configuration files used throughout OpenSearch. Sometimes alternatively called parameters, the programming language manipulating the key-value pair usually dictates the name of this mapping in a YAML file. For OpenSearch documentation (Java), they are properly a `Setting` object.
```

**Review by natebower — APPROVED** [review]
@cwillum One minor change. Otherwise, LGTM.

**Pull request merged**